PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Shipped.txt

```diff
@@ -108,6 +113,7 @@ Microsoft.ApplicationInsights.DataContracts.ExceptionTelemetry.ExceptionTelemetr
 Microsoft.ApplicationInsights.DataContracts.ExceptionTelemetry.ExceptionTelemetry(System.Exception exception) -> void
 Microsoft.ApplicationInsights.DataContracts.ExceptionTelemetry.HandledAt.get -> Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt
 Microsoft.ApplicationInsights.DataContracts.ExceptionTelemetry.HandledAt.set -> void
+Microsoft.ApplicationInsights.DataContracts.ExceptionTelemetry.ExceptionDetailsInfoList.get -> System.Collections.Generic.IReadOnlyList<Microsoft.ApplicationInsights.DataContracts.ExceptionDetailsInfo>
```

Comment:
I'd suggest to make it a `list` so customer can add new exception details and remove ones they are not interested in

Reply:
I thought about that, but intentionally didn't add for few reasons:
1. If user wants to add a new `ExceptionDetailsInfo` object, how would he/she create the internal `ExceptionDetails` object? Currently `ExceptionDetailsInfo` is purely designed to be invoked from within AI and not from outside.

2. `ExceptionDetails` hold the relation between parent-child exception objects using `id` and `outerId`. If we add or remove it, it will be tricky to reflect those changes on actual parent-child relation present in corresponding internal `ExceptionDetails` object. We don't want to mess up that relation by giving user too much control (at least thats what my thinking is).

Thoughts?

---
In reply to: [203859148](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/865#discussion_r203859148) [](ancestors = 203859148)